pkg/awsutils/awsutils.go

```diff
@@ -1168,23 +1174,26 @@ func (cache *EC2InstanceMetadataCache) GetENILimit() (int, error) {
 		info := output.InstanceTypes[0]
 		// Ignore any missing values
 		instanceType := aws.StringValue(info.InstanceType)
-		eniLimit = int(aws.Int64Value(info.NetworkInfo.MaximumNetworkInterfaces))
-		ipLimit := int(aws.Int64Value(info.NetworkInfo.Ipv4AddressesPerInterface))
-		if instanceType != "" && eniLimit > 0 && ipLimit > 0 {
-			InstanceENIsAvailable[instanceType] = eniLimit
-			InstanceIPsAvailable[instanceType] = ipLimit
+		eniLimit := aws.Int64Value(info.NetworkInfo.MaximumNetworkInterfaces)
+		ipv4Limit := aws.Int64Value(info.NetworkInfo.Ipv4AddressesPerInterface)
+		if instanceType != "" && eniLimit > 0 && ipv4Limit > 0 {
+			eniLimits = InstanceTypeLimits{
+				ENILimit:  eniLimit,
+				IPv4Limit: ipv4Limit,
+			}
+			InstanceNetworkingLimits[instanceType] = eniLimits
 		} else {
 			return 0, errors.New(fmt.Sprintf("%s: %s", UnknownInstanceType, cache.instanceType))
 		}
 	}
-	return eniLimit, nil
+	return int(eniLimits.ENILimit), nil
```

Comment:
We should pick an int size for this and use it consistently (`int` or `int64`) - to avoid overflow issues.  I imagine `int` is more than sufficient for the number of ENIs (typically O(10)), and IP addresses (which is either going to be O(100) or effectively limitless (eg IPv6 /64 subnet)).

Reply:
I guess the main issue here is that the EC2 API returns these values..
```
eniLimit := aws.Int64Value(info.NetworkInfo.MaximumNetworkInterfaces)
ipv4Limit := aws.Int64Value(info.NetworkInfo.Ipv4AddressesPerInterface)
```
I would prefer using `int` everywhere, but I guess the question is where we should convert. 😃

Reply:
Oh right, the in64 came from the API :(
In that case, we should probably convert to int as soon as possible (so it's unambiguous after that point):

```golang
eniLimit := int(aws.Int64Value(info.NetworkInfo.MaximumNetworkInterfaces))
ipv4Limit := int(aws.Int64Value(info.NetworkInfo.Ipv4AddressesPerInterface))
```